lib/ecto/adapters/sql.ex

```diff
@@ -38,19 +38,15 @@ defmodule Ecto.Adapters.SQL do
       ## Types
 
       @doc false
-      def autogenerate(:id), do: nil
-      def autogenerate(:embed_id), do: Ecto.UUID.autogenerate()
-      def autogenerate(:binary_id), do: Ecto.UUID.autogenerate()
+      def autogenerate(:id),        do: nil
+      def autogenerate(:embed_id),  do: Ecto.UUID.generate()
+      def autogenerate(:binary_id), do: Ecto.UUID.bingenerate()
 
       @doc false
-      def loaders({:embed, _} = type, _), do: [&Ecto.Adapters.SQL.load_embed(type, &1)]
-      def loaders(:binary_id, type), do: [Ecto.UUID, type]
-      def loaders(_, type), do: [type]
+      defdelegate loaders(primitive, type), to: Ecto.Adapters.SQL
 
       @doc false
-      def dumpers({:embed, _} = type, _), do: [&Ecto.Adapters.SQL.dump_embed(type, &1)]
-      def dumpers(:binary_id, type), do: [type, Ecto.UUID]
-      def dumpers(_, type), do: [type]
+      defdelegate dumpers(primitive, type), to: Ecto.Adapters.SQL
```

Comment:
Let's not delegate and rather inline as it was before. These functions are called *a lot*.